hardware/pico_driver/src/pico_driver.cc

```diff
@@ -430,10 +432,21 @@ class PicoDriverL2 : public PicoDriver, public royale::IExtendedDataListener {
       ROS_WARN("data pointer = nullptr");
       return;
     }
+
+    ros::Time commonStamp(0, 0);
+    if (data->hasDepthData() && data->getDepthData() != nullptr) {
+      commonStamp.fromNSec(
+        std::chrono::duration_cast<std::chrono::nanoseconds>(data->getDepthData()->timeStamp).count());
+    } else if (data->hasIntermediateData()
+               && data->getIntermediateData() != nullptr) {
+      commonStamp.fromNSec(
+        std::chrono::duration_cast<std::chrono::nanoseconds>(data->getIntermediateData()->timeStamp).count());
+    }
+
     // If we have depth data, use the same mechanism as L1 to push it
     if (data->hasDepthData() && pub_cloud_.getNumSubscribers() > 0
       && data->getDepthData() != nullptr) {
-      cloud_.header.stamp = ros::Time::now();
+      cloud_.header.stamp = commonStamp;
```

Comment:
why not instead of:
cloud_.header.stamp = commonStamp;

do:
cloud_.header.stamp.fromNSec(
        std::chrono::duration_cast<std::chrono::nanoseconds>(data->getDepthData()->timeStamp).count());

seems that we could compact this instead of creating a new variable